CMakeLists.txt

```diff
@@ -41,6 +41,10 @@ option(DLAF_WITH_HIP "Enable HIP support" OFF)
 cmake_dependent_option(
   DLAF_WITH_MPI_GPU_SUPPORT "Enable MPI GPU support" OFF "DLAF_WITH_CUDA OR DLAF_WITH_HIP" OFF
 )
+cmake_dependent_option(
+  DLAF_WITH_MPI_GPU_SUPPORT_FORCE_CONTIGUOUS "Force GPU buffers to be contigous before communicating" ON
```

Comment:
```suggestion
  DLAF_WITH_MPI_GPU_SUPPORT_FORCE_CONTIGUOUS "Force GPU buffers to be contiguous before communicating" ON
```

Reply:
Is there any rationale behind the decision of having it defaulted to ON? I know benchmark results, just wondering if it is somehow future-proof or not...

Reply:
No rationale other than what you already mentioned. The parent option `DLAF_WITH_MPI_GPU_SUPPORT` is anyway off by default so I thought this doesn't hurt. I would consider the whole thing experimental and for that reason I don't feel too bad about defaulting `DLAF_WITH_MPI_GPU_SUPPORT_FORCE_CONTIGUOUS` to on without more benchmarking results.

Reply:
https://github.com/eth-cscs/DLA-Future/pull/1096/commits/1afdf643f33bf8a27b071611f480fc83b0aef4b1 and bonus: https://github.com/eth-cscs/DLA-Future/pull/1096/commits/d5bc59e221bdbbcf7021ad79856d5afd3bbbb663.